### PR TITLE
Gracefully handle interruptions in the Helix logic.

### LINKED ISSUE #1763: [Failed CI Test] testClusterStatusMonitorLifecycle(org.apache.helix.monitoring.TestClusterStatusMonitorLifecycle)
This issue is created for tracking unstable test: testClusterStatusMonitorLifecycle(org.apache.helix.monitoring.TestClusterStatusMonitorLifecycle)

### PR DESCRIPTION
**Pull request by jiajunwang**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

Fixes #1763 

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

This PR aims to address the problem that the interrupt signal is swallowed by the catch logic so the thread is not really interrupted. This may cause leakage or race conditions.

### Tests

- [X] The following tests are written for this issue:

New test TestAsyncBaseStage has been added to verify that AbstractAsyncBaseStage in the async thread can be canceled correctly.
The other changes are already covered by the existing tests to prove that they won't fail the normal logic.

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

**Review by junkaixue — APPROVED** [review]
lgtm

**Comment by jiajunwang** [comment]
> lgtm, but seems your build failed?

Tried again and it works. I didn't find any clue like Exceptions. Moreover, I run this test in my localhost for 50 times, very stable. I think we can check-in the PR for now since we agree the change makes sense.
I will follow up if this test becomes unstable after merging.

**Comment by jiajunwang** [comment]
Approved by @junkaixue , I will merge after https://github.com/apache/helix/pull/1778 since they are related.

**Pull request merged**
